src/models/os.ts

```diff
@@ -348,40 +349,52 @@ const getOsModel = function (
 
 	const _getAllOsVersions = async (
 		deviceTypes: string[],
-		options?: PineOptions<Release>,
+		options: PineOptions<Release> | undefined,
+		convenienceFilter: 'supported' | 'include_draft' | 'all',
 	): Promise<Dictionary<OsVersion[]>> => {
-		const hostapps = await _getOsVersions(deviceTypes, options);
+		const extraFilterOptions =
```

Comment:
primarily just moved from bellow and replaced the `=== 'all` check with `convenienceFilter === 'supported' || convenienceFilter === 'include_draft'`